src/System.ComponentModel.Composition/tests/Microsoft/Internal/Collections/ReadOnlyDictionaryTests.cs

```diff
@@ -0,0 +1,206 @@
+            {
+                dictionary.Remove("Value");
+            });
+        }
+
+        [Fact]
+        public void Remove2_ShouldThrowNotSupported()
+        {
+            var dictionary = GetReadOnlyDictionaryWithData();
+
+            Assert.Throws<NotSupportedException>(() =>
+            {
+                dictionary.Remove(new KeyValuePair<string, object>("Key", "Value"));
+            });
+        }
+
+        [Fact]
+        public void ItemSet_ShouldThrowNotSupported()
+        {
+            var dictionary = GetReadOnlyDictionaryWithData();
+
+            Assert.Throws<NotSupportedException>(() =>
+            {
+                dictionary["Key"] = "Value";
+            });
+        }
+
+        [Fact]
+        public void Keys_ShouldReturnWrappedDictionaryKeys()
+        {
+            var dictionary = GetWritableDictionaryWithData();
+            var readOnlyDictionary = GetReadOnlyDictionary(dictionary);
+
+            Assert.Equal(readOnlyDictionary.Keys, dictionary.Keys);
+        }
+
+        [Fact]
+        public void Values_ShouldReturnWrappedDictionaryValues()
+        {
+            var dictionary = GetWritableDictionaryWithData();
+            var readOnlyDictionary = GetReadOnlyDictionary(dictionary);
+
+            Assert.Equal(readOnlyDictionary.Values, readOnlyDictionary.Values);
+        }
+
+        [Fact]
+        public void IsReadOnly_ShouldAlwaysBeTrue()
+        {
+            var dictionary = GetWritableDictionaryWithData();
+            var readOnlyDictionary = GetReadOnlyDictionary(dictionary);
+            
+            Assert.False(dictionary.IsReadOnly);
+            Assert.True(readOnlyDictionary.IsReadOnly);
+        }
+
+        [Fact]
+        public void Count_ShouldReturnWrappedDictionaryCount()
+        {
+            var dictionary = GetWritableDictionaryWithData();
+            var readOnlyDictionary = GetReadOnlyDictionary(dictionary);
+
+            Assert.Equal(dictionary.Count, readOnlyDictionary.Count);
+        }
+
+        [Fact]
+        public void ContainsKey()
+        {
+            var dictionary = GetWritableDictionaryWithData();
+            var readOnlyDictionary = GetReadOnlyDictionary(dictionary);
+
+            Assert.True(readOnlyDictionary.ContainsKey("Key1"));
+            Assert.False(readOnlyDictionary.ContainsKey("InvalidKey"));
+        }
+
+        [Fact]
+        public void Contains()
+        {
+            var dictionary = GetWritableDictionaryWithData();
+            var readOnlyDictionary = GetReadOnlyDictionary(dictionary);
+
+            Assert.True(readOnlyDictionary.Contains(new KeyValuePair<string,object>("Key1", "Value1")));
+            Assert.False(readOnlyDictionary.Contains(new KeyValuePair<string,object>("InvalidKey", "Value1")));
+            Assert.False(readOnlyDictionary.Contains(new KeyValuePair<string,object>("Key1", "InvalidValue")));
+        }
+
+        [Fact]
+        public void CopyTo()
```

Comment:
Should we also iterate through the data to compare that both have the same items?